packages/app-market/src/components/apps-installed/apps-installed-suggested.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { FC, useMemo } from 'react'
+import { isTruthy } from '@reapit/utils-common'
+import { InstalledSuggestedImage, InstalledSuggestedImageWrapper } from './__styles__'
+import { elFadeIn, MediaType, Subtitle, useMediaQuery } from '@reapit/elements'
+import { AppsBrowseConfigCollection, useAppsBrowseState } from '../../core/use-apps-browse-state'
+import { Carousel } from '../carousel'
+import { navigate } from '../../utils/navigation'
+import { Routes } from '../../constants/routes'
+import { useHistory } from 'react-router-dom'
+
+export const handleCarouselCols = (mediaQuery: MediaType) => () => {
+  const { isMobile, isTablet, isDesktop } = mediaQuery
+
+  if (isMobile) {
+    return 1
+  }
+
+  if (isTablet || isDesktop) {
+    return 2
+  }
+
+  return 3
+}
+
+export const handleSortConfigs = (appsBrowseConfigState: AppsBrowseConfigCollection | null) => () => {
+  const featuredHeroApps =
+    appsBrowseConfigState?.data.filter((config) => config.configType === 'featuredHeroApps') ?? []
+  const heroApps = appsBrowseConfigState?.data.filter((config) => config.configType === 'heroApps') ?? []
+
+  const featuredHeroAppDetails = featuredHeroApps
+    .map(({ content, filters }) => {
```

Comment:
you can use reduce to minimize null returns 

```ts
featured = featuredHeroApps.reduce((array, { content, filters }) => {
  if (content?. imageUrl && Array.isArray(filters.id) && filters.id.length >= 1) {
     array.push({ imageUrl: content.imageUrl, id: filters.id[0], })
  }
}, [])
```

Reasons: in your method, there's 3 loops, the initial map, the flat and the filter func.

Reply:
I tried that, firstly there are only 2 items in the array plus it's memoised so not expensive.

Also, none of the above typechecks, to actually make it work you have to type cast all over the place, set unhelpful defaults and things that will make the result less reliable.

In the end I chose reducing the risk of runtime errors over a small performance gain given we are only iterating over two items.